### PR TITLE
DHIS2 API version

### PR DESCRIPTION
**Pull request by kaapstorm**
##### SUMMARY
This PR changes the way we handle DHIS2 API versioning.

Originally we thought it best to write for a specific version of the API (version 26, which is the version given in all examples in [their documentation](https://dhis2.github.io/dhis2-api-specification/)). 

When a `Dhis2Repeater` makes API calls, it gives the API version in the URL, e.g. ".../api/26/events" for the Events endpoint.

Unfortunately API calls that request version 26 are no longer supported by current stable versions of DHIS2. It responds with 404 errors. However, the API features and schema have not changed much (if at all) since version 26.

The approach taken by this PR is to use the version of the DHIS2 instance to work out what features it supports. Currently CommCare/Motech uses features that have been present since API version 15, and we have not integrated with a server older than that.

If we want to offer features only available in newer API versions, `Dhis2Repeater.api_version` will allow us to check whether the remote system supports the feature or not.

Review by commit :blowfish: 

##### FEATURE FLAG
DHIS2 Integration

### CODE REVIEW

File: corehq/motech/dhis2/repeaters.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def api_version(self):
         e.g. Not all CRUD operations are supported before version 15.
         """
         if self.dhis2_version:
-            return api_version_re.match(self.dhis2_version).groups(1)
```
Comment (esoergel) [inline corehq/motech/dhis2/repeaters.py]: using `Version` or `StrictVersion` would help this out too

**Review by esoergel — COMMENTED** [review]
I do still think it'd be prudent to pick a maximum known supported version, done in such a way that it's super easy to bump.  Otherwise, if this always attempts to interact with the latest version - any breaking changes could potentially cause issues as soon as they're rolled out.

File: corehq/motech/dhis2/repeaters.py [hunk 2]
```diff
@@ -15,10 +17,14 @@
...
-from corehq.motech.repeaters.views.repeaters import AddDhis2RepeaterView
 from corehq.motech.requests import Requests
 from corehq.toggles import DHIS2_INTEGRATION
-from corehq.util import reverse
+
+
+def is_dhis2_version(value):
+    if re.match(r'^2\.\d+(\.\d)?$', value):
+        return True
+    raise BadValueError(_('Value must be a DHIS2 version in the format "2.xy" or "2.xy.z".'))
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/motech/dhis2/repeaters.py]: Nice! I'll do that.

**Comment by kaapstorm** [comment]
> it'd be prudent to pick a maximum known supported version, done in such a way that it's super easy to bump.

@karmuno suggested the same thing. I think it's a good idea.

File: corehq/motech/dhis2/repeaters.py [hunk 3]
```diff
@@ -91,15 +93,16 @@ def get_payload(self, repeat_record):
         return json.loads(payload)
 
     @property
-    def api_version(self):
+    def api_version(self) -> int:
```
Comment (esoergel) [inline corehq/motech/dhis2/repeaters.py]: :star2:

**Review by esoergel — APPROVED** [review]
looks great!

**Pull request merged**